client/app.py

```diff
@@ -88,20 +90,30 @@ def main():
     return render_template('home.html', 
                             characters=list(game_const.CHARACTERS))
 
-@APP.route('/join', methods=['POST'])
+@APP.route('/join_game', methods=['POST'])
 def join_game():
-    selected_character = request.form.get('character')
-    logging.info('Selected character: %s', selected_character)
-    if selected_character:
-        join_response = APP.app_data.server.send_join_request(selected_character)
-        logging.info("Client ID: %s", join_response.client_id)
-        APP.characters.append(join_response.player)
-        APP.client_ids.append(join_response.client_id)
-
-    # Add logic here to determine if game is ready
+    # Player selects a character
+    APP.app_data.character = request.form.get('character')
+    logging.info('Selected character: %s', APP.app_data.character)
+
+    # Player get client id  
+    join_response = APP.app_data.server.send_join_request(APP.app_data.character)
+    logging.info("Client ID: %s", join_response.client_id)
+    APP.app_data.client_id = join_response.client_id
+
+    # Player get game id
+    game_response = APP.app_data.server.send_start_game_request()
+    logging.info("Game ID: %s", join_response.client_id)
+    APP.app_data.game_id = game_response.game_id
+
+    # Player request game state
+    client = Client(APP.app_data.character, SERVER_IP, SERVER_PORT, APP.app_data.game_id)
+    game_state = client.send_game_state(list(game_const.CHARACTERS), APP.app_data.character)
```

Comment:
send_game_state expects a list of Players and a character which I think I am passing correctly here.

Reply:
Isn't client boundary for the server to use, and server boundary is for the client to use?